zilliqa/src/consensus.rs

```diff
@@ -202,6 +204,8 @@ pub struct Consensus {
     db: Arc<Db>,
     /// Transactions ordered by priority, map of address of TXn (from account) to ordered TXns to be executed.
     new_transactions_priority: BTreeMap<Address, BinaryHeap<TxnOrder>>,
+    /// PRNG
+    rng: ChaCha8Rng,
```

Comment:
Consensus has a PRNG that is seeded from 0. This makes the tests fully deterministic again, and since they are using the prng modulo the committee minus themselves, their behaviour shouldn't be identical every time. Open to seeding it with the secret key too.